### PR TITLE
Documentation gathering updates when resolving tickets

### PR DESCRIPTION
**Pull request by willwatters**
### Change description ###
Documentation gathering updates when resolving tickets.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -577,20 +551,62 @@ app.view('document_help_request', async ({ ack, body, view, client }) => {
...
+        const helpRequestMessages = (await client.conversations.replies({
+            channel: reportChannelId,
+            ts: body.view.private_metadata,
+            limit: 200, // after a thread is 200 long we'll break but good enough for now
+        })).messages
+
+        if (helpRequestMessages.length > 0 && (
+            helpRequestMessages[0].text === 'New platform help request raised' ||
+            helpRequestMessages[0].text === 'Duplicate issue')
+        ) {
```
Comment (timja) [inline app.js:563]: is there a reason you added this check?

File: app.js [hunk 2]
```diff
@@ -577,20 +551,62 @@ app.view('document_help_request', async ({ ack, body, view, client }) => {
...
+            await client.chat.update({
+                channel: reportChannelId,
+                ts: body.view.private_metadata,
+                text: 'New platform help request raised',
+                blocks: blocks
+            });
+        
+            //console.log(JSON.stringify(body, null, 2));
+            //console.log(body.view.state.values.category_block.category.selected_option)
+
```
Comment (timja) [inline app.js:593]: remove commented out code when you submit it
```suggestion
```

File: src/messages.js [hunk 3]
```diff
@@ -766,6 +753,8 @@ function helpRequestDocumentation({where, what, how}) {
     ];
 }
 
+
+
```
Comment (timja) [inline src/messages.js:757]: remove unneeded whitespace
```suggestion
```

File: src/service/persistence.js [hunk 4]
```diff
@@ -141,10 +141,12 @@ function convertEmail(email) {
         return systemUser
     }
 
-    return email.split('@')[0]
+    //TODO: justice.gov emails no longer match 1:1 with jura usernames, so this may fail post migration to justice.gov
```
Comment (timja) [inline src/service/persistence.js:144]: maybe something like this, it should probably be in a separate pull request though
```suggestion
    // TODO: justice.gov emails no longer match 1:1 with Jira usernames, so this may fail post migration to justice.gov.uk, this hack mostly works but if there are multiple people with the same name this could cause issues.
```

File: src/service/persistence.js [hunk 5]
```diff
@@ -141,10 +141,12 @@ function convertEmail(email) {
...
+    return email.split('@')[0].replace(/[0-9]/g, '')
 }
 
 async function createHelpRequestInJira(summary, project, user, labels) {
+    console.log(`Creating help request in jura for user: ${user}`)
```
Comment (timja) [inline src/service/persistence.js:149]:
```suggestion
    console.log(`Creating help request in Jira for user: ${user}`)
```

File: src/service/persistence.js [hunk 6]
```diff
@@ -211,12 +213,37 @@ async function addCommentToHelpRequest(externalSystemId, fields) {
     }
 }
 
+async function addCommentToHelpRequestResolve(externalSystemId, { category, how} ) {
```
Comment (timja) [inline src/service/persistence.js:216]: would you be able to add a test for this function and the addLabel function to https://github.com/hmcts/slack-help-bot/blob/main/src/service/persistence.functional.test.js

we run those when Jira is getting upgraded to make sure everything still works. and it's quicker to develop with those tests than by running through the slack-help-bot all the time

**Review by timja — COMMENTED** [review]
Looks really good, few suggestions and would be great if we could get a couple of tests, let me know if you need a hand on this

File: app.js (hunk 1 above)
Comment (willwatters) [reply app.js]: Have removed this

File: app.js (hunk 2 above)
Comment (willwatters) [reply app.js]: Commented out code removed

File: src/messages.js (hunk 3 above)
Comment (willwatters) [reply src/messages.js]: Space removed

File: src/service/persistence.js (hunk 6 above)
Comment (willwatters) [reply src/service/persistence.js:216]: Test functions added for addCommentToHelpRequestResolve and addLabel functions and tested using jest.

File: src/messages.js [hunk 7]
```diff
@@ -683,13 +683,12 @@ function resolveHelpRequestBlocks({thread_ts}) {
                         "emoji": true
                     },
                     "options": [
-                        option('Infrastructure issue'),
-                        option('Lack of documentation'),
+                        option('External (Github/Azure/Sonacloud) issue'),
```
Comment (timja) [inline src/messages.js:686]:
```suggestion
                        option('External (GitHub/Azure/SonarCloud) issue'),
```

**Review by timja — COMMENTED** [review]
Does / work in the label? have you tested it?

File: src/messages.js (hunk 7 above)
Comment (willwatters) [reply src/messages.js:686]: Yes I tested this and it works ok - see https://tools.hmcts.net/jira/browse/SBOX-310

Comment (willwatters) [reply src/messages.js]: Typo corrected:

option('External (GitHub/Azure/SonarCloud) issue'),

File: app.js [hunk 8]
```diff
@@ -577,20 +551,56 @@ app.view('document_help_request', async ({ ack, body, view, client }) => {
...
+            ts: body.view.private_metadata,
+            limit: 200, // after a thread is 200 long we'll break but good enough for now
+        })).messages
 
-        const documentation = {
-            what: body.view.state.values.what_block.what.value,
-            where: body.view.state.values.where_block.where.value,
-            how: body.view.state.values.how_block.how.value,
-        };
+        {
```
Comment (timja) [inline app.js:560]:
```suggestion
```

File: app.js [hunk 9]
```diff
@@ -577,20 +551,56 @@ app.view('document_help_request', async ({ ack, body, view, client }) => {
...
+            
+            await addLabel(jiraId, documentation)
+
+            await client.chat.postMessage({
+                channel: reportChannel,
+                thread_ts: body.view.private_metadata,
+                text: 'Platform help request documented',
+                blocks: helpRequestDocumentation(documentation)
+            });
+        }
```
Comment (timja) [inline app.js:603]:
```suggestion
```

**Review by timja — APPROVED** [review]
seems to be a random block here that's unneeded? (and formatting changes to adjust)

File: app.js (hunk 8 above)
Comment (willwatters) [reply app.js]: Removed unnecessary block and reformatted.

File: app.js (hunk 9 above)
Comment (willwatters) [reply app.js]: Removed unnecessary block and reformatted.
